code/game/objects/structures/stool_bed_chair_nest/alien_nests.dm

```diff
@@ -56,6 +63,7 @@
 	lock_atom(M, /datum/locking_category/bed/nest)
 	src.add_fingerprint(user)
 	overlays += image(icon,"nest-covering",MOB_LAYER)
+	random_key = rand(9999)
```

Comment:
I believe "resist" uses do_after (which checks every few seconds if you've been stunned/are still on the chair/still cuffed) so there are no perfect solutions.